### ISSUE TITLE
Standalone route handlers are not copied upon registration in router

### ISSUE THREAD

**Issue by jtraub** [starlite-api/starlite#703]
**Describe the bug**
[Documentation](https://starlite-api.github.io/starlite/usage/1-routing/4-registering-components-multiple-times/#registering-standalone-route-handlers-multiple-times) mentions that handler functions are copied upon registration within a router. It seems that copying happens only for controller route handlers.

As the result weird behavior appears when user relies on layering.

**To Reproduce**
Consider the following code

```python
from starlite import BaseRouteHandler, MediaType, Request, Router, Starlite, get


def guard1(request: Request, _: BaseRouteHandler) -> None:
    print("route guard")


def guard2(request: Request, _: BaseRouteHandler) -> None:
    print("router guard")


def guard3(request: Request, _: BaseRouteHandler) -> None:
    print("router wrapper guard")


def guard4(request: Request, _: BaseRouteHandler) -> None:
    print("app guard")


@get(path="/test", media_type=MediaType.TEXT, guards=[guard1])
def test() -> str:
    return "ok"


router = Router(path="/router", route_handlers=[test], guards=[guard2])

router_wrapper = Router(path="/wrapper", route_handlers=[router], guards=[guard3])

app = Starlite(
    debug=True,
    route_handlers=[test, router_wrapper],
    guards=[guard4],
)
```

Launching it with uvicorn and visiting `/wrapper/router/test` doesn't trigger `guard2` and `guard3`.
```
./venv/bin/uvicorn test:app
INFO:     Started server process [671196]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
app guard
route guard
INFO:     127.0.0.1:55792 - "GET /wrapper/router/test HTTP/1.1" 200 OK
```

**Comment by peterschutt** [comment]
Good pickup - also not matching the behavior [defined in guards docs](https://starlite-api.github.io/starlite/usage/9-guards/?h=layer#guard-scopes):

> This means you can add multiple guards at every layer. Unlike dependencies, guards do not override each other but are rather cumulative. This means that you can define guards on different levels of your app, and they will combine.

```python
test_route_map = app.route_map["/test"]
wrapper_router_test_route_map = app.route_map["/wrapper/router/test"]
test_route_map_handler = test_route_map["_asgi_handlers"]["GET"]["handler"]
wrapper_router_test_route_map_handler = app.route_map["/wrapper/router/test"]["_asgi_handlers"]["GET"]["handler"]
print(test_route_map_handler.guards == wrapper_router_test_route_map_handler.guards)  # True
```

Also reverse the order of registration and `/test` route gets all guards applied.

**Comment by peterschutt** [comment]
Although to clarify my comment, the way the guards resolve is fine, reversing the registration order shows that are accumulating as they should.

So this is prob a better verification of the issue:

```python
test_route_map = app.route_map["/test"]
wrapper_router_test_route_map = app.route_map["/wrapper/router/test"]
test_route_map_handler = test_route_map["_asgi_handlers"]["GET"]["handler"]
wrapper_router_test_route_map_handler = app.route_map["/wrapper/router/test"]["_asgi_handlers"]["GET"]["handler"]
print(test_route_map_handler is wrapper_router_test_route_map_handler)
```

**Comment by peterschutt** [comment]
closed in #704
